packages/reqresp/test/utils/response.ts

```diff
@@ -0,0 +1,33 @@
+import {pipe} from "it-pipe";
+import {allForks} from "@lodestar/types";
+import {responseEncodeError, responseEncodeSuccess} from "../../src/encoders/responseEncode.js";
+import {RespStatus} from "../../src/interface.js";
+import {EncodedPayload, EncodedPayloadType, ProtocolDefinition} from "../../src/types.js";
+import {ResponseChunk} from "../fixtures/encoders.js";
+import {beaconConfig} from "../fixtures/messages.js";
+import {blocksToReqRespBlockResponses} from "../utils/block.js";
+import {arrToSource} from "../utils/index.js";
+
+export async function* responseEncode(
+  responseChunks: ResponseChunk[],
+  protocol: ProtocolDefinition<any, any>
+): AsyncIterable<Buffer> {
+  for (const chunk of responseChunks) {
+    if (chunk.status === RespStatus.SUCCESS) {
+      if (chunk.payload.type === EncodedPayloadType.bytes) {
+        return [chunk.payload.bytes];
+      }
+
+      const lodestarResponseBodies = protocol.method.startsWith("beacon_blocks")
+        ? blocksToReqRespBlockResponses(([chunk.payload.data] as unknown) as allForks.SignedBeaconBlock[], beaconConfig)
+        : [chunk.payload];
```

Comment:
This lines look really nasty the fact that this burried utility uses `beacon_blocks` prefix to switches logic. The `EncodedPayload` should be passed as an argument to this function, and decided by the test caller if use bytes or ssz.

Reply:
I can refactor that part to better approach, but for the PR scope it was intended to reuse existing tests, so why this ported as it is.

Reply:
oh okay